components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/APIControllerUtil.java

```diff
@@ -0,0 +1,804 @@
+            defaultProductionEndpoint.addProperty(ImportExportConstants.ENDPOINT_URL,
+                    ImportExportConstants.DEFAULT_DYNAMIC_ENDPOINT_URL);
+            defaultSandboxEndpoint.addProperty(ImportExportConstants.ENDPOINT_URL,
+                    ImportExportConstants.DEFAULT_DYNAMIC_ENDPOINT_URL);
+            updatedDynamicEndpointParams.addProperty(ImportExportConstants.ENDPOINT_TYPE_PROPERTY,
+                    ImportExportConstants.DEFAULT_DYNAMIC_ENDPOINT_URL);
+            updatedDynamicEndpointParams
+                    .addProperty(ImportExportConstants.FAILOVER_ROUTING_POLICY, Boolean.FALSE.toString());
+            handleEndpointValues(endpointsObject, updatedDynamicEndpointParams, defaultProductionEndpoint,
+                    defaultSandboxEndpoint);
+            //add dynamic endpoint configs as endpoint configs
+            return updatedDynamicEndpointParams;
+
+        } else if (ImportExportConstants.AWS_TYPE_ENDPOINT.equals(endpointType)) {// if endpoint type is AWS Lambda
+            //if aws config is not provided
+            if (envParams.get(ImportExportConstants.AWS_LAMBDA_ENDPOINT_JSON_PROPERTY) == null) {
+                throw new APIManagementException("Please specify awsLambdaEndpoints field for  and continue...");
+            }
+            JsonObject awsEndpointParams = envParams.get(ImportExportConstants.AWS_LAMBDA_ENDPOINT_JSON_PROPERTY)
+                    .getAsJsonObject();
+            JsonObject updatedAwsEndpointParams = new JsonObject();
+            //if the access method is provided with credentials
+            if (StringUtils
+                    .equals(awsEndpointParams.get(ImportExportConstants.AWS_ACCESS_METHOD_JSON_PROPERTY).getAsString(),
+                            ImportExportConstants.AWS_STORED_ACCESS_METHOD)) {
+                //get the same config object for aws configs
+                updatedAwsEndpointParams = awsEndpointParams;
+                updatedAwsEndpointParams.remove(ImportExportConstants.AWS_ACCESS_METHOD_JSON_PROPERTY);
+                updatedAwsEndpointParams.addProperty(ImportExportConstants.AWS_ACCESS_METHOD_PROPERTY,
+                        ImportExportConstants.AWS_STORED_ACCESS_METHOD);
+            } else {
+                //if the credentials are not provided the default will be used
+                updatedAwsEndpointParams.addProperty(ImportExportConstants.AWS_ACCESS_METHOD_PROPERTY,
+                        ImportExportConstants.AWS_ROLE_SUPPLIED_ACCESS_METHOD);
+            }
+            updatedAwsEndpointParams.addProperty(ImportExportConstants.ENDPOINT_TYPE_PROPERTY,
+                    ImportExportConstants.AWS_LAMBDA_TYPE_ENDPOINT);
+            handleEndpointValues(endpointsObject, updatedAwsEndpointParams, defaultProductionEndpoint,
+                    defaultSandboxEndpoint);
+            //add AWS endpoint configs as endpoint configs
+            return updatedAwsEndpointParams;
+        }
+        return null;
```

Comment:
Shall we throw an exception here instead of returning null? Since we only support either Dynamic or AWS Endpoint type, any other value could lead to a NPE.